frontend/packages/console-app/src/components/detect-perspective/__tests__/PerspectiveDetector.spec.tsx

```diff
@@ -67,7 +68,28 @@ describe('PerspectiveDetector', () => {
     );
     mockPerspectives[1].properties.usePerspectiveDetection = () => testPromise;
 
-    (useExtensions as jest.Mock).mockImplementation(() => mockPerspectives);
+    (usePerspectives as jest.Mock).mockImplementation(() => mockPerspectives);
+
+    const wrapper = mount(<PerspectiveDetector setActivePerspective={setActivePerspective} />);
+    await act(async () => {
+      promiseResolver(() => [false, false]);
+    });
+    expect(wrapper.isEmptyRender()).toBe(true);
+    expect(setActivePerspective).toHaveBeenCalledWith('admin');
+  });
+
+  it('should set admin as default perspective if all perspectives are disabled', async () => {
+    window.SERVER_FLAGS.perspectives =
+      '[{ "id" : "dev", "visibility": {"state": "AccessReview" , "accessReview": { "missing" : [{ "resource" : "namespaces", "verb" : "list" }]}}}, { "id" : "dev-test", "visibility": {"state" : "Disabled" }}, { "id" : "admin", "visibility": {"state" : "Disabled" }}]';
```

Comment:
Sorry for such nit picking. But the test description (all are disabled) and test cases (1 has access review and two are disabled) don't match here, or?

Btw, you might can write this for the tests instead:

```tsx
const visibility: PerspectiveVisibility[] = [
  {},
  {},
  {},
];
window.SERVER_FLAGS.perspectives = JSON.stringify(visibility);
```